src/connectionManager.js

```diff
@@ -544,43 +542,51 @@ export default class ConnectionManager {
             // Clone the array
             const credentials = credentialProvider.credentials();
 
-            return Promise.all([findServers()]).then(responses => {
+            return Promise.all([findServers()]).then((responses) => {
 
                 const foundServers = responses[0];
                 let servers = credentials.Servers.slice(0);
                 mergeServers(credentialProvider, servers, foundServers);
-
                 servers.sort((a, b) => (b.DateLastAccessed || 0) - (a.DateLastAccessed || 0));
                 credentials.Servers = servers;
                 credentialProvider.credentials(credentials);
-
                 return servers;
             });
         };
 
         function findServers() {
-
             return new Promise((resolve, reject) => {
-
-                const onFinish = foundServers => {
-                    const servers = foundServers.map(foundServer => {
-
-                        const info = {
+                var onFinish = function (foundServers) {
+                    var servers = foundServers.map((foundServer) => {
+                        var info = {
                             Id: foundServer.Id,
-                            LocalAddress: convertEndpointAddressToManualAddress(foundServer) || foundServer.Address,
-                            Name: foundServer.Name
+                            LocalAddress:
+                                convertEndpointAddressToManualAddress(
+                                    foundServer
+                                ) || foundServer.Address,
+                            Name: foundServer.Name,
                         };
-
-                        info.LastConnectionMode = info.ManualAddress ? ConnectionMode.Manual : ConnectionMode.Local;
-
+                        info.LastConnectionMode = info.ManualAddress
+                            ? ConnectionMode.Manual
+                            : ConnectionMode.Local;
                         return info;
                     });
                     resolve(servers);
                 };
 
-                serverDiscoveryFn.findServers(1000).then(onFinish, () => {
-                    onFinish([]);
-                });
+                if (
+                    window.NativeShell &&
+                    typeof window.NativeShell.findServers === "function"
+                ) {
```

Comment:
I kind of hate this change, we'll have to find a fix once the API client is fully functional. We definitely shouldn't fragment the NativeShell usage even further.

Reply:
This was in the current API Client.

I had to copy-paste this function in, as it straight up didn't work.

I agree that it's nasty, though.

Reply:
Yeah, I figured as much. I was just mentally bookmarking this section for the eventual NativeShell refactor to improve all our wrapper clients.